frontend/app/src/services/converters.ts

```diff
@@ -139,3 +139,13 @@ export function convertVaultDetails(
     events: convertVaultEvents(details.events)
   }));
 }
+
+export function deserializeApiErrorMessage(
+  message: string
+): { [key: string]: string[] } | undefined {
+  try {
+    return JSON.parse(message.replace(/'/gi, '"'));
```

Comment:
What's this doing?

Reply:
The message is stringified json but instead of escaped double quotes it has single quotes. Single quotes are not valid json, so before parsing I need to replace them. If parsing fails it returns undefined.

Reply:
Ah I see. This is the backend's fault though and should be fixed, instead of having hacks in the frontend.

I believe it's because this function naively stringifies any error from the validation:

https://github.com/rotki/rotki/blob/32ae260146373014c1957ff8f7c2508a1a46acac/rotkehlchen/api/server.py#L152-L162

Instead you could do something like this:

```diff
diff --git a/rotkehlchen/api/server.py b/rotkehlchen/api/server.py
index 50d2c0b6..279c7db7 100644
--- a/rotkehlchen/api/server.py
+++ b/rotkehlchen/api/server.py
@@ -1,3 +1,4 @@
+import json
 import logging
 from http import HTTPStatus
 from typing import Any, Dict, List, Optional, Tuple, Union
@@ -159,7 +160,15 @@ def handle_request_parsing_error(
 ) -> None:
     """ This handles request parsing errors generated for example by schema
     field validation failing."""
-    abort(HTTPStatus.BAD_REQUEST, result=None, message=str(err))
+    msg = str(err)
+    if isinstance(err.messages, dict):
+        # first key is just the location. Ignore
+        key = list(err.messages.keys())[0]
+        msg = json.dumps(err.messages[key])
+    elif isinstance(err.messages, list):
+        msg = ','.join(err.messages)
+
+    abort(HTTPStatus.BAD_REQUEST, result=None, message=msg)
```
which essentially interprets the error messages differently if they are a dict, list or anything else.

And if they are a dict, it serializes it as proper json with `json.dumps`.


Note: Instead of fixing this here we can also do this in another issue, as many error checking tests that await single `'` in test messages will now have to be changed to `"`.

Reply:
Did it myself. Would be nice though if you could check my removal of the code to not replace single with double quotes. Maybe I did something wrong there @kelsos

Reply:
ok let me check